conans/search/search.py

```diff
@@ -90,11 +90,19 @@ def search_recipes(paths, pattern=None, ignorecase=True):
         for subdir in subdirs:
             conan_ref = ConanFileReference(*subdir.split("/"))
             if pattern:
-                if pattern.match(str(conan_ref)):
+                if _partial_match(pattern, conan_ref):
                     ret.append(conan_ref)
+
         return sorted(ret)
 
 
+def _partial_match(pattern, conan_ref):
+    tokens = str(conan_ref).replace('/', ' / ').replace('@', ' @ ').split()
+
+    partial_sums = reduce(lambda c, x: c + [c[-1] + x], tokens, [''])[1:]
```

Comment:
This reads a bit complex in my opinion, it would be nice to have it simplified, or maybe commented to explain what is doing.

Reply:
I agree

Reply:
I see it now since you mentioned it :sweat_smile: . Sorry, I've started to do stuff in Python only recently. I'll look into making things clear a bit later, thanks!

Reply:
Yes, that is one of the dangerous things of python, as it is quite expressive, it allows to write very compact code, which is nice, but might affect readability. Quite an impressive code if you started to do stuff in Python recently 😄